src/Hazelcast.Net/Serialization/SerializationService.Lookup.cs

```diff
@@ -100,6 +101,9 @@ private ISerializerAdapter LookupKnownSerializer(Type type)
         {
             // fast path for some known serializers
 
+            if (typeof(CompactGenericRecordBase).IsAssignableFrom(type))
+                return _compactSerializerAdapter;
+
```

Comment:
We can also add this as an `||` to the if below.

Reply:
yes - it's intentional, makes it easier to read the different cases IMO